cider-dap/src/main.rs

```
@@ -135,10 +138,10 @@ where
     };
 
     // Open file using the extracted program path
-    let file = File::open(program_path)?;
+    // let file = File::open(program_path)?;
 
     // Construct the adapter
-    let mut adapter = MyAdapter::new(file);
+    let mut adapter = MyAdapter::new(program_path).unwrap();
```

```suggestion
    let mut adapter = MyAdapter::new(program_path)?;
```

---

Better to propagate the error rather than explode